dbms/src/Functions/FunctionsJson.h

```diff
@@ -1514,4 +1514,464 @@ class FunctionJsonDepth : public IFunction
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 2; }
+
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments[0]->onlyNull() || arguments[1]->onlyNull())
+            return makeNullable(std::make_shared<DataTypeNothing>());
+
+        if unlikely (!removeNullable(arguments[0])->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        if unlikely (!removeNullable(arguments[1])->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        return makeNullable(std::make_shared<DataTypeString>());
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const auto & json_col = block.getByPosition(arguments[0]).column;
+        const auto & path_col = block.getByPosition(arguments[1]).column;
+        if (json_col->onlyNull() || path_col->onlyNull())
+        {
+            block.getByPosition(result).column
+                = block.getByPosition(result).type->createColumnConst(block.rows(), Null());
+            return;
+        }
+
+        auto nested_block = createBlockWithNestedColumns(block, arguments);
+        auto json_source = createDynamicStringSource(*nested_block.getByPosition(arguments[0]).column);
+        auto path_source = createDynamicStringSource(*nested_block.getByPosition(arguments[1]).column);
+
+        auto col_to = ColumnString::create();
+        auto & data_to = col_to->getChars();
+        auto & offsets_to = col_to->getOffsets();
+        size_t rows = block.rows();
+        offsets_to.resize(rows);
+        auto col_null_map = ColumnUInt8::create(rows, 0);
+        auto & vec_null_map = col_null_map->getData();
+
+        if (json_col->isColumnNullable())
+        {
+            const auto & json_column_nullable = static_cast<const ColumnNullable &>(*json_col);
+            if (path_col->isColumnNullable())
+            {
+                const auto & path_column_nullable = static_cast<const ColumnNullable &>(*path_col);
+                doExecuteCommon<true, true>(
+                    json_source,
+                    json_column_nullable.getNullMapData(),
+                    path_source,
+                    path_column_nullable.getNullMapData(),
+                    rows,
+                    data_to,
+                    offsets_to,
+                    vec_null_map);
+            }
+            else if (path_col->isColumnConst())
```

Comment:
If `json_col` is constant and `path_col` is not constant, will `doExecuteCommon` handle it correctly?

Reply:
yes, ut added here.
https://github.com/pingcap/tiflash/pull/8491/files#diff-f4a535f55a6fc34ac53f3aeaeb0d5df7f0a7a57812bcfb062bf9aad74eb4984aR98-R107